### PR TITLE
feat(pie-icons) - DSW-881 remove fill attr from icons

### PR DESCRIPTION
**Pull request by dandel10n**
## Describe your changes (can list changeset entries if preferable)

The name for the following social icons has been changed to reflect their static behaviour(they are multicoloured and shouldn't change colour depending on the background they are on):

- android-static and - android-static-large
- apple-static and apple-static-large
- blogger-static and  blogger-static-large
- facebook-static and facebook-static-large
- google-static and google-static-large
- instagram-static and instagram-static-large
- linkedin-static and linkedin-static-large
- microsoft-static and microsoft-static-large
- microsoft-circle-filled-static and microsoft-circle-filled-static-large
- microsoft-circle-static and microsoft-circle-static-large
- pinterest-static and pinterest-static-large
- twitter-static and twitter-static-large
- youtube-static and youtube-static-large

- Changed pie-icons build to remove fill attr from icon paths and add `fill="currentColor"` for svg elements avoiding multicoloured icons.
- Rebuilt pie-icons-react and pie-icons-vue after the pie-icons update
- Updated doc site with the new named icons
- Readme update for pie-icons, pie-icons-react and pie-icons-vue libraries

Docs site feature url: https://pr528.dvskdcoepjoyf.amplifyapp.com/foundations/iconography/library/

### CODE REVIEW

File: packages/tools/pie-icons-vue/README.md [hunk 1]
```diff
@@ -86,17 +84,17 @@ The component extends [@justeat/browserslist-config-fozzie](https://github.com/j
...
 
 Icons should be added as SVGs to the main pie-icons package and published, before simply incrementing the dependency of `pie-icons` in the `pie-icons-vue` package, to generate the new set of Vue components.
 
-The PIE iconset is managed by our PIE design team and new icon requests should go through them to ensure that they are designed inline with our standards and guildelines. Please reach out to the #proj-pie-design-system team who can help with these requests.
+The PIE iconset is managed by our PIE design team and new icon requests should go through them to ensure that they are designed inline with our standards and guildelines. Please reach out to PIE design system team using #help-designsystem slack channel.
 
 ### Building the Module
 
-Run `yarn build` to compile the module.
+Run `yarn turbo run build --filter=pie-icons-vue` from the root level to compile the module.
```
Comment (ashleynolan) [inline packages/tools/pie-icons-vue/README.md:97]:
```suggestion
Run `yarn build --filter=pie-icons-vue` from the root level to compile the module.
```

File: packages/tools/pie-icons-react/README.md [hunk 2]
```diff
@@ -66,21 +71,21 @@ import AlertTriangle from "@justeattakeaway/pie-icons-react/dist/AlertTriangle";
...
 
 ### Building the Module
 
-Run `yarn build` to compile the module.
+Run `yarn turbo run build --filter=pie-icons-react` from the root level to compile the module.
 
 ## Icon list
 
-You can check the list of all the icons in our [Storybook](https://react.pie.design/?path=/story/design-pie-icons--icons-story).
+You can check the list of all the icons on our PIe documentation site: https://pie.design/foundations/iconography/library/.
```
Comment (fernandofranca) [inline packages/tools/pie-icons-react/README.md:88]: Tiny typo `on our PIE`

**Review by ashleynolan — COMMENTED** [review]
Few comments from me

**Review by fernandofranca — COMMENTED** [review]
Left a couple of comments

File: packages/tools/pie-icons-react/README.md [hunk 3]
```diff
@@ -22,12 +26,13 @@ yarn add @justeattakeaway/pie-icons-react
 
 ```tsx
 import { IconAlertTriangleLarge, IconCalendar } from "@justeattakeaway/pie-icons-react";
+import { styles as globalStyles } from '@ds-library/src/styles';
```
[the remark this answers is not in the document]
Comment (dandel10n) [reply packages/tools/pie-icons-react/README.md:29]: I added this to show that you can pass fill attr with pie alias colour token. But example is related specifically to stacks. I will update it to fill="PIE_ALIAS_COLOUR_ToKEN" instead.

File: packages/tools/pie-icons-react/README.md [hunk 4]
```diff
@@ -66,21 +71,21 @@ import AlertTriangle from "@justeattakeaway/pie-icons-react/dist/AlertTriangle";
...
 
 Icons should be added as SVGs to the main pie-icons package and published, before simply incrementing the dependency of `pie-icons` in the `pie-icons-react` package, to generate the new set of Vue components.
 
-The PIE iconset is managed by our PIE design team and new icon requests should go through them to ensure that they are designed inline with our standards and guildelines. Please reach out to the #proj-pie-design-system team who can help with these requests.
+The PIE iconset is managed by our PIE design system team and new icon requests should go through them to ensure that they are designed inline with our standards and guildelines. Please reach out to PIE design system team using #help-designsystem slack channel.
 
 ### Building the Module
 
-Run `yarn build` to compile the module.
+Run `yarn turbo run build --filter=pie-icons-react` from the root level to compile the module.
```
[the remark this answers is not in the document]
Comment (dandel10n) [reply packages/tools/pie-icons-react/README.md:84]: Hmm, when `yarn build --filter=pie-icons-react` I get `Unbound variable "TURBO_TOKEN"` but when I `run turbo run build --filter=pie-icons-react` it builds fine

File: packages/tools/pie-icons/bin/process-svgs.js [hunk 5]
```diff
@@ -12,15 +12,15 @@ console.log(`Processing SVGs in ${IN_DIR}...`);
 const svgFiles = getAllSvgs().filter((file) => path.extname(file.fileName) === '.svg');
 
 svgFiles.forEach((svgObject) => {
-    const fullPath = path.join(svgObject.path, '/', svgObject.fileName);
+    const normalisedFilename = (svgObject.fileName).toLowerCase();
```
[the remark this answers is not in the document]
Comment (dandel10n) [reply packages/tools/pie-icons/bin/process-svgs.js:15]: Good catch! Do you know why some of the icons start from uppercase letter?

File: packages/tools/pie-icons-react/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (dandel10n) [reply packages/tools/pie-icons-react/README.md]: Got it, I updated to show both running from project level and root level

**Review by raoufswe — APPROVED** [review]

File: packages/tools/pie-icons/bin/process-svgs.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ashleynolan) [reply packages/tools/pie-icons/bin/process-svgs.js]: They hopefully shouldn't (it's be human error if they are named that way in the Google Drive), but the mistake in naming happened before, which is why we now have this check 😆

File: apps/pie-docs/src/iconData.json [hunk 6]
```diff
@@ -1697,8 +1697,8 @@
             "displayName": "Social",
             "icons": [
                 {
-                    "name": "facebook",
-                    "displayName": "Facebook"
+                    "name": "facebook-static",
+                    "displayName": "Facebook Static"
```
Comment (ashleynolan) [inline apps/pie-docs/src/iconData.json:1701]: Docs site question – do we want the display names to have static in them? (not sure if you strip that out, but may be better to strip it out when the icons get compiled)

**Review by ashleynolan — COMMENTED** [review]
Tiny comment left from me – otherwise LGTM

File: packages/tools/pie-icons-react/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ashleynolan) [reply packages/tools/pie-icons-react/README.md]: Weirdly, it runs with `yarn build --filter=pie-icons-react` from the project root for me. Not sure if it's just my setup though – may be worth checking with someone else as well and see what works for them too!

**Review by fernandofranca — APPROVED** [review]

File: apps/pie-docs/src/iconData.json (hunk 6 above)
Comment (dandel10n) [reply apps/pie-docs/src/iconData.json:1701]: I didn't. Feels like a good indication for expected icon behaviour: that static variants won't change colour like non-static. For example, `facebook-circle` can change colour on hover/focus or during theme change etc while `facebook-static` can't.

File: apps/pie-docs/src/_11ty/shortcodes/categorisedIconList.js [hunk 7]
```diff
@@ -12,9 +12,18 @@ const pieIcons = require('../filters/pieIconsSvg')();
 const buildIconCard = (icon) => {
     const pieIcon = pieIcons.find((x) => x.name === icon.name).icon;
```
Comment (xander-marjoram) [inline apps/pie-docs/src/_11ty/shortcodes/categorisedIconList.js:13]: I think this line will need optional chaining too
```suggestion
    const pieIcon = pieIcons.find((x) => x.name === icon.name)?.icon;
```

File: packages/tools/pie-icons/src/__tests__/icon.test.js [hunk 8]
```diff
@@ -24,10 +24,10 @@ describe('toSvg()', () => {
         expect(icon1.toSvg()).toMatchSnapshot();
     });
     test('appends color and stroke-width attributes when passed through', () => {
```
Comment (xander-marjoram) [inline packages/tools/pie-icons/src/__tests__/icon.test.js:26]:
```suggestion
    test('appends fill and stroke-width attributes when passed through', () => {
```

File: apps/pie-docs/src/iconData.json (hunk 6 above)
Comment (ashleynolan) [reply apps/pie-docs/src/iconData.json:1701]: Sounds good - I don't mind either way. May be worth just flagging with @jordifornellsJET that it'll change the icon names like this once this gets merged

Comment (dandel10n) [reply apps/pie-docs/src/iconData.json:1701]: I talked to Molly already. Will let Jordi know as well.

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**